### ISSUE TITLE
Usage of django.ImageField leads to fixture 'dict' not found

### ISSUE THREAD

**Issue by sliverc** [pytest-dev/pytest-factoryboy#58]
Using a factory with a django.Image field leads to error `fixture 'dict' not found`.

Sample:
```python
from factory.django import DjangoModelFactory, ImageField
from pytest_factoryboy import register

from . import models

class AttachmentFactory(DjangoModelFactory):
    path = ImageField(width=1024, height=768)

    class Meta:
        model = models.Attachment

register(AttachmentFactory)

# this test will lead to `fixture 'dict' not found` error
def test_attachment(attachment):
  # some tests
```

This happens with following dependencies:
```
factory-boy==2.10.0
pytest-factoryboy==2.0.1
```

This worked without any problems with these dependencies:
```
factory-boy==2.8.1
pytest-factoryboy==1.3.2
```

**Comment by olegpidsadnyi** [comment]
@sliverc what's models.Attachment? is it just a dict?

**Comment by sliverc** [comment]
No it is a simple model.

```python
class Attachment(models.Model):
    path = models.ImageField(max_length=1024)
```

I think the problem is caused through a factory boy change where `ImageField` resp. `FileField` is now derived from a `declarations.Dict` class.

See https://github.com/FactoryBoy/factory_boy/blob/master/factory/django.py#L175

**Comment by olegpidsadnyi** [comment]
@sliverc I see.. wow.. in factoryboy 2.10.0 ImageField is a SubFactory now, more than that - subclass of a Dict, this what brings the "dict" as a name of it's model. Not sure what kind of sense does it make.

Initially my idea was that SubFactory points to a factory of a model, for which you need a fixture.

**Comment by olegpidsadnyi** [comment]
Basically this:

```python

import factory
from pytest_factoryboy import register


class Foo(dict):
    pass


@register
class FooFactory(factory.Factory):

    class Meta:
        model = Foo

    bar = factory.declarations.Dict({})


def test_foo(foo):
    """this test will lead to `fixture 'dict' not found` error."""
    pass

```

**Comment by olegpidsadnyi** [comment]
@sliverc We could fix it by comparing a type of the declaration. This would have an implication that people can't subclass SubFactory or RelatedFactory to refer to other models anymore.

Let's add an issue in factory-boy and ask what do they think why ImageField has to be a SubFactory.

**Comment by sliverc** [comment]
I also don't quite see why it needs to be a subclass of SubFactory. Let's see what upstream has to say I guess.

**Comment by noirbizarre** [comment]
Hi !

Is there any news or workaround for this one ?

**Comment by olegpidsadnyi** [comment]
@noirbizarre have you tried? I see Rafael has closed the issue https://github.com/FactoryBoy/factory_boy/issues/451

**Comment by olegpidsadnyi** [comment]
@sliverc is it still an issue? can be closed?

**Comment by sliverc** [comment]
I have tested it with the factory boy master and it works as expected. Thanks.

The only problem is that the factory boy change is not released yet so factory boy git master needs to be added as dependency for this to work.

I am fine with that and this issue can be closed. Question is whether others will still stumble upon this issue till factory boy has an official release with this fix.

**Comment by noirbizarre** [comment]
I tried but on latest public release. If this is working with the master, it's fine to me and I'll wait for the release

**Comment by sliverc** [comment]
Factory Boy has been released and I have tested it with version `2.11.1`. It works so closing this issue. @olegpidsadnyi Thanks for having worked on this.

**Comment by MRigal** [comment]
@sliverc This is actually still problematic when using `factory.Dict` and `factory.List`, as they are also subfactories... :-/
